homeassistant/components/aseko_pool_live/config_flow.py

```diff
@@ -60,21 +65,74 @@ async def async_step_user(
                 _LOGGER.exception("Unexpected exception")
                 errors["base"] = "unknown"
             else:
-                await self.async_set_unique_id(info[CONF_UNIQUE_ID])
-                self._abort_if_unique_id_configured()
+                return await self.async_store_credentials(info)
 
-                return self.async_create_entry(
-                    title=info[CONF_EMAIL],
-                    data={CONF_ACCESS_TOKEN: info[CONF_ACCESS_TOKEN]},
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.data_schema,
+            errors=errors,
+        )
+
+    async def async_store_credentials(self, info: dict[str, Any]) -> ConfigFlowResult:
+        """Store validated credentials."""
+
+        if self.reauth_entry:
+            self.hass.config_entries.async_update_entry(
+                self.reauth_entry,
+                title=info[CONF_EMAIL],
+                data={
+                    CONF_EMAIL: info[CONF_EMAIL],
+                    CONF_PASSWORD: info[CONF_PASSWORD],
+                },
+            )
+            await self.hass.config_entries.async_reload(self.reauth_entry.entry_id)
+            return self.async_abort(reason="reauth_successful")
+
+        await self.async_set_unique_id(info[CONF_UNIQUE_ID])
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=info[CONF_EMAIL],
+            data={
+                CONF_EMAIL: info[CONF_EMAIL],
+                CONF_PASSWORD: info[CONF_PASSWORD],
+            },
+        )
+
+    async def async_step_reauth(
+        self, user_input: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauth upon an API authentication error."""
+
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+
+        return await self.async_step_reauth_confirm(user_input)
```

Comment:
We probably shouldn't pass the existing config entry data as user input directly to the confirm step.